main.go

```diff
@@ -26,11 +30,28 @@ func main() {
 		log.Fatal(err)
 	}
 
+	sigs := make(chan os.Signal, 1)
+	abort := make(chan bool, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		sig := <-sigs
+		io.WriteString(os.Stderr, fmt.Sprintf("Aborting! Caught signal \"%s\"\n", sig))
+		io.WriteString(os.Stderr, "Cleaning up...\n")
+		abort <- true
+		go func() {
```

Comment:
something about this goroutine immediately inside another goroutine smells.

does something like:

```go
select {
  case abort <- true:
  case <- time.After(10 * time.Second):
    fmt.Fprintf(os.Stderr, "Taking too long....")
    os.Exit(1)
}
```

work just as well?